django/db/backends/oracle/base.py

```diff
@@ -239,17 +241,66 @@ def __init__(self, *args, **kwargs):
         )
         self.features.can_return_columns_from_insert = use_returning_into
 
+    @property
+    def is_pool(self):
+        return self.settings_dict["OPTIONS"].get("pool", False)
+
+    @property
+    def pool(self):
+        if not self.is_pool:
+            return None
+
+        if self.settings_dict.get("CONN_MAX_AGE", 0) != 0:
+            raise ImproperlyConfigured(
+                "Pooling doesn't support persistent connections."
+            )
+
+        pool_key = (self.alias, self.settings_dict["USER"])
+        if pool_key not in self._connection_pools:
+            connect_kwargs = self.get_connection_params()
+            pool_options = connect_kwargs["pool"]
+
+            if pool_options is not True:
+                for key, val in pool_options.items():
+                    connect_kwargs[key] = val
+
+            del connect_kwargs["pool"]
+
+            pool = Database.create_pool(
+                user=self.settings_dict["USER"],
+                password=self.settings_dict["PASSWORD"],
+                dsn=dsn(self.settings_dict),
+                **connect_kwargs,
+            )
+            self._connection_pools.setdefault(pool_key, pool)
+
+        return self._connection_pools[pool_key]
+
+    def close_pool(self):
+        pool_key = (self.alias, self.settings_dict["USER"])
+        if self.pool:
+            self.pool.close(force=True)
+            del self._connection_pools[pool_key]
```

Comment:
There is no need to create `pool_key` when it's not used:
```suggestion
        if self.pool:
            self.pool.close(force=True)
            pool_key = (self.alias, self.settings_dict["USER"])
            del self._connection_pools[pool_key]
```

Reply:
Done.